sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/AmqpUtil.java

```diff
@@ -90,6 +95,11 @@ public static int getDataSerializedSize(Message amqpMessage) {
     }
 
     private static int sizeof(Object obj) {
+        if (obj == null) {
+            // In AMQP, null is a special type which has no data beyond the encoding byte.
+            return 0;
+        }
+
```

Comment:
For null value scenario, is the sizeof API returning illegal argument error?

Reply:
Yes, that's where the NPE was throwing from: the first instanceof was trying to reference the object and throwing.